lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -799,4 +844,397 @@ export default class Recipe extends RecipeModule {
+              primaryUserId: string;
+              description: string;
+          }
+        | {
+              status: "RECIPE_USER_ID_ALREADY_LINKED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR";
+              user: User;
+          }
+    > {
+        let primaryUser: User;
+        let targetUser: User;
+        // We first try and check which one is a primary user
+        if (user1.isPrimaryUser) {
+            // If both of the are primary, then they cannot be linked
+            // This is a condition that should be checked before calling this helper, but it could be some kind of race-condition
+            if (user2.isPrimaryUser) {
+                return { status: "BOTH_USERS_PRIMARY" };
+            }
+            primaryUser = user1;
+            targetUser = user2;
+        } else if (user2.isPrimaryUser) {
+            // If we got here we know not both of them are primary
+            // so we can just use user2 as primary
+            primaryUser = user2;
+            targetUser = user1;
+        } else {
+            // We select the older user
+            const olderUser = user1.timeJoined >= user2.timeJoined ? user2 : user1;
+            const newerUser = olderUser === user1 ? user2 : user1;
+            // Try and make it primary first
+            let shouldMakeOlderUserPrimary = await this.shouldBecomePrimaryUser(
+                olderUser,
+                tenantId,
+                session,
+                userContext
+            );
+            if (shouldMakeOlderUserPrimary) {
+                const createPrimaryUserResult = await this.recipeInterfaceImpl.createPrimaryUser({
+                    recipeUserId: olderUser.loginMethods[0].recipeUserId,
+                    userContext,
+                });
+                if (createPrimaryUserResult.status !== "OK") {
+                    return createPrimaryUserResult;
+                }
+                primaryUser = createPrimaryUserResult.user;
+                targetUser = newerUser;
+            } else {
+                // if the older one shouldn't become primary, we try the newer one
+                const shouldMakeNewerUserPrimary = await this.shouldBecomePrimaryUser(
+                    newerUser,
+                    tenantId,
+                    session,
+                    userContext
+                );
+                if (shouldMakeNewerUserPrimary) {
+                    const createPrimaryUserResult = await this.recipeInterfaceImpl.createPrimaryUser({
+                        recipeUserId: newerUser.loginMethods[0].recipeUserId,
+                        userContext,
+                    });
+                    if (createPrimaryUserResult.status !== "OK") {
+                        return createPrimaryUserResult;
+                    }
+                    primaryUser = createPrimaryUserResult.user;
+                    targetUser = olderUser;
+                } else {
+                    // If this also return false, we cannot link
```

Comment:
this could also come here cause of email verification related issues..

Reply:
Yeah, but the point of we cannot link is that at this point the users are not primary and we can't make any of them primary, so we can't link.
The comment is explaining why we do not continue, but I'll expand it.